solutions/cses-1163.mdx

```diff
@@ -61,47 +62,151 @@ int main(){
 }
 ```
 
-## Solution 2 - Reversing the steps
-
-I found another solution by reversing all the steps and calculating the maximum
-length at each step. Here is my implementation.
-
+</CPPSection>
+</LanguageSection>
+
+	
+## Solution 2 - Going Backwards
+
+**Time Complexity:** $\mathcal{O}(n)$
+
+We'll start by trying to find the maximum gap once all the traffic lights are
+added. This is the last thing we'll output, so we'll add it to the end of our
+output array. Then, we'll remove traffic lights in the reverse order to how they
+were added, and find the gap each removal creates.
+
+This gap is just the distance between the two street coordinates (either a
+traffic light or the beginning or ending of the street) next to a given traffic
+light stored in our set, so we can use our set to find these values and subtract
+them to get our gap.
+
+But this gap may not be the maximum gap. We'll compare this to the gap we found
+once all the traffic lights are added, and set the max gap to the greater value.
+We'll then set the next lowest element in the output array to this value, which
+will represent the greatest gap prior to adding the traffic light we just
+removed.
+
+<LanguageSection>
+<JavaSection>
+
+<Warning>
+We have to use some nonstandard io (input/output) to pass all test cases in the allowed time on CSES, because of its slow grader. 
+</Warning>
+```java
+import java.io.*;
+import java.util.*;
+
+public class TrafficLights {
+	public static void main(String[] args) throws IOException {
+		BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
+		PrintWriter pw = new PrintWriter(System.out);
+		StringTokenizer st = new StringTokenizer(r.readLine());
+
+		int streetLength = Integer.parseInt(st.nextToken());
+		int lightNum = Integer.parseInt(st.nextToken());
+
+		// Using an array to read values since we can't get values from sets in Java
+		int[] opArray = new int[lightNum];
+		TreeSet<Integer> streetPositions = new TreeSet<>();
```

Comment:
polymorphism

Reply:
Just out of curiosity-why?

Reply:
a good java practice.

Reply:
uh harry can i have an so link that says to do that

Reply:
I think he's talking about the 3rd bullet point [here](https://usaco.guide/general/adding-solution/#java). BTW do you think my PRs are good enough to make content author or will I have to make some better ones

Reply:
i have no clue lol idk what the standards are

Reply:
![image](https://user-images.githubusercontent.com/55369003/129412725-b4d85503-53c3-40b0-ab8d-7624c31f4a71.png)
you mean this one?

Reply:
![image](https://user-images.githubusercontent.com/40570437/129412829-1589d21b-bf7e-40bf-bb8f-27fa13fde36d.png)
this one